src/main/java/com/github/emilshina/utils/JsonParserUtils.java

```diff
@@ -0,0 +1,33 @@
+package com.github.emilshina.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Array;
+
+/**
+ * Parser for json files for deserialization.
+ */
+public final class JsonParserUtils {
+
+    private JsonParserUtils() {
+        throw new UnsupportedOperationException("Illegal access to private constructor.");
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T[] fromJsonAsArray(final Gson gson, final JsonElement json, final Class<T> tClass,
```

Comment:
you need to revise this logic to accept a class which could be either array or a single item... and convert a single item to array if required...